### PR TITLE
CI: do not reuse builds on release branches

### PR DESCRIPTION
**Pull request by maxknv**
#do_not_test

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 7200a5a51e547dbf74cbc82ef3469a06fd1b8e82 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59798/7200a5a51e547dbf74cbc82ef3469a06fd1b8e82/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59798/7200a5a51e547dbf74cbc82ef3469a06fd1b8e82/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: .github/workflows/release_branches.yml [hunk 1]
```diff
@@ -236,7 +243,7 @@ jobs:
       - name: Check out repository code
         uses: ClickHouse/checkout@v1
         with:
-          clear-repository: true
+          filter: tree:0
```
Comment (Felixoid) [inline .github/workflows/release_branches.yml:246]: I think we can remove the whole `with` here, filter is meaningless with default `depth: 0`

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -318,15 +318,21 @@ jobs:
...
+      - BuilderBinDarwin
+      - BuilderBinDarwinAarch64
+      - BuilderDebRelease
+      - BuilderDebAarch64
+    runs-on: [self-hosted, style-checker]
+    steps:
+      - name: Check out repository code
+        uses: ClickHouse/checkout@v1
+        with:
+          filter: tree:0
```
Comment (Felixoid) [inline .github/workflows/master.yml:331]:
```suggestion
```

let's use default checkout

File: tests/ci/ci_config.py [hunk 3]
```diff
@@ -828,9 +826,6 @@ def validate(self) -> None:
         ),
     },
     other_jobs_configs={
-        JobNames.MARK_RELEASE_READY: TestConfig(
```
Comment (Felixoid) [inline tests/ci/ci_config.py:831]: It just removes the logic for `Ready for release` from ci.py, and we will control it through the workflow yml files, right?

**Review by Felixoid — COMMENTED** [review]
LGTM

File: tests/ci/ci_config.py (hunk 3 above)
Comment (maxknv) [reply tests/ci/ci_config.py:831]: yeap

**Review by Felixoid — APPROVED** [review]

**Pull request merged**